File: odoo-19.0+e.20260101/odoo/addons/website_helpdesk_livechat/static/tests/mock_server/models/res_users.js
import { livechatModels } from "@im_livechat/../tests/livechat_test_helpers";

import { serverState } from "@web/../tests/web_test_helpers";

export class ResUsers extends livechatModels.ResUsers {
    _init_store_data(store) {
        super._init_store_data(...arguments);
        store.add({
            helpdesk_livechat_active: true,
            has_access_create_ticket: this.env.user?.group_ids.includes(
                serverState.groupHelpdeskId
            ),
        });
    }
}

File: odoo-19.0+e.20260101/odoo/addons/website_helpdesk_livechat/models/res_users.py
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models
from odoo.addons.mail.tools.discuss import Store


class ResUsers(models.Model):
    _inherit = "res.users"

    def _init_store_data(self, store: Store):
        super()._init_store_data(store)
        domain = [("use_website_helpdesk_livechat", "=", True), ('company_id', 'in', self.env.context.get('allowed_company_ids', []))]
        store.add_global_values(
            helpdesk_livechat_active=self.env["helpdesk.team"].sudo().search_count(domain) > 0,
            has_access_create_ticket=self.env.user.has_group("helpdesk.group_helpdesk_user"),
        )
